steve/util.py

```diff
@@ -216,18 +391,21 @@ def get_project_config_file_name():
     return SteveConfig.file_name
 
 
-def create_project_config_file(dir_path):
+def create_project_config_file(dir_path, cfg=None):
```

Comment:
I am not so familiar with those (as you might have guessed), but it does help you prevent passing in referenced objects there and get "unexpected" results:

    def abc(x, y=[]):
        y.append(x)
        print(y)
    
    abc(1)
    abc(2)
    
gives different print out from

    def abc(x, y=None):
        if y is None:
            y = []
        y.append(x)
        print(y)
    
    abc(1)
    abc(2)